backend/test/v2/test_utils.go

```diff
@@ -218,3 +218,21 @@ func DeleteAllPipelines(client *api_server.PipelineClient, t *testing.T) {
 		assert.True(t, isRemoved)
 	}
 }
+
+func GetPipelineUploadClient(
+	uploadPipelinesWithKubernetes bool,
+	isKubeflowMode bool,
+	isDebugMode bool,
+	namespace string,
+	clientConfig clientcmd.ClientConfig,
+) (api_server.PipelineUploadInterface, error) {
+	if uploadPipelinesWithKubernetes {
```

Comment:
optional: given this is essentially a Factory pattern, this feels like it should be a enum rather than a series of bool switches. ie take in mode=`(kubeflow|kubernetes|default)`.

Reply:
I'll skip this for now since it would involve a lot more refactoring in the rest of the test code.